packages/perf/lib/index.js

```diff
@@ -47,12 +50,40 @@ class FirebasePerfModule extends FirebaseModule {
   constructor(...args) {
     super(...args);
     this._isPerformanceCollectionEnabled = this.native.isPerformanceCollectionEnabled;
+    this._instrumentationEnabled = this.native.isInstrumentationEnabled;
   }
 
   get isPerformanceCollectionEnabled() {
     return this._isPerformanceCollectionEnabled;
   }
 
+  get instrumentationEnabled() {
+    return this._instrumentationEnabled;
+  }
+
+  set instrumentationEnabled(enabled) {
+    if (!isBoolean(enabled)) {
+      throw new Error("firebase.perf().instrumentationEnabled = 'enabled' must be a boolean.");
+    }
+    if (Platform.OS == 'ios') {
+      // We don't change for android as it cannot be set from code, it is set at gradle build time.
+      this._instrumentationEnabled = enabled;
+      // No need to await, as it only takes effect on the next app run.
+      this.native.instrumentationEnabled(enabled);
+    }
+  }
+
+  get dataCollectionEnabled() {
+    return this._isPerformanceCollectionEnabled;
+  }
+
+  set dataCollectionEnabled(enabled) {
+    if (!isBoolean(enabled)) {
+      throw new Error("firebase.perf().dataCollectionEnabled = 'enabled' must be a boolean.");
+    }
+    this.setPerformanceCollectionEnabled(enabled);
```

Comment:
And now, irritatingly because I think I was trying to de-duplicate and *asked* for this to delegate, I don't think this can call through because the non-JS-API `this.setPerformanceCollectionEnabled` needs to toggle both to match historical behavior

I was tempted to be a bit cynical and question if anyone was even using this but it is actually a vital part of a correct GDPR opt-in flow and I am motivated to respect user privacy so...no cynicism, just effort 😆